apps/web/src/app/dashboard/teacher/quiz/_components/NewQuizButton.tsx

```diff
@@ -44,43 +28,45 @@ import {
   CardFooter,
   CardHeader
 } from "@/components/ui/card";
-import { trpc } from "../../../../../utils/trpc/client";
 import { useState } from "react";
-import { TrpcReactQueryOptions } from "../../../../../utils/trpc/lib";
 import Image from "next/image";
 import { Heading } from "src/components/ui";
+import { trpc } from "../../../../../utils/trpc/client";
 import { ScrollArea } from "@/components/ui/scroll-area";
 import { Separator } from "@/components/ui/separator";
 
 interface Props {
-  initialData: TrpcReactQueryOptions["questionBank"]["list"]["initialData"];
+  initialData: {
+    questionsCount: number;
+    _count: {
+      questions: number;
+    };
+    id: number;
+    createdAt: Date;
+    updatedAt: Date;
+    title: string;
+    authorId: string;
+  };
 }
 interface quizProps {
   no_of_questions: number;
-  question_bank_id: number;
   quiz_name: string;
   timer: number;
 }
 export const NewQuizButton = ({ initialData }: Props) => {
+  console.log(initialData);
   const [isOpen, setOpen] = useState<boolean>(false);
   const [initializeQuiz, setInitializeQuiz] = useState<boolean>(false);
   const [quizSettings, setQuizSettings] = useState<quizProps>({
     no_of_questions: 0,
-    question_bank_id: 0,
     quiz_name: "",
     timer: 0
   });
-  const [maxQns, setMaxQns] = useState<number>(0);
-  const { data: questionsBanks } = trpc.questionBank.list.useQuery(undefined, {
-    initialData
-  });
+  const [maxQns] = useState<number>(initialData.questionsCount);
```

Comment:
state is never set, so do we need a state hook? why not just replace maxQns with initialData.questionsCount